test/tempfile.h

```diff
@@ -52,6 +52,7 @@ class TempFile
   }
 
   TempFile(const TempFile& ) = delete;
+  void operator=(const TempFile& ) = delete;
```

Comment:
Hum.. I would say that this commit should be merged with a previous one, but the commit message raise a question here.

If we allow a copy assignment, we end with two TempFile on the same fd, and so two `close` of the fd. Why do you need to allow copy assignment ?

Reply:
Copy assignment is not allowed. It is explicitly forbidden. The commit message states the same in different words (i.e. before this commit there was a loophole enabling to copy the TempFile object, now that loophole is closed).

Reply:
Ho.. I totally miss read that. I was thinking you were removing the deletion of the copy-assignment. And obviously this is the contrary. My bad. We are good here.

Reply:
Merged this commit into the first one.